packages/lesswrong/server/repos/PostEmbeddingsRepo.ts

```diff
@@ -0,0 +1,34 @@
+import AbstractRepo from "./AbstractRepo";
+import PostEmbeddings from "../../lib/collections/postEmbeddings/collection";
+import { randomId } from "../../lib/random";
+
+export default class PostEmbeddingsRepo extends AbstractRepo<DbPostEmbedding> {
+  constructor() {
+    super(PostEmbeddings);
+  }
+
+  setPostEmbeddings(
+    postId: string,
+    postHash: string,
+    model: string,
+    embeddings: number[],
+  ): Promise<null> {
+    const now = new Date();
+    return this.none(`
+      INSERT INTO "PostEmbeddings" (
+        "_id",
+        "postId",
+        "postHash",
+        "lastGeneratedAt",
+        "embeddings",
+        "model",
+        "createdAt"
+      ) VALUES (
+        $1, $2, $3, $4, $5, $6, $4
+      ) ON CONFLICT ("postId", "model") DO UPDATE SET
```

Comment:
Just checking: these are in fact set to be unique keys?

Reply:
I'm not sure that they are? Will this ON CONFLICT ever run?

Reply:
We have a unique index on postId and model (`ensureIndex(PostEmbeddings, {postId: 1, model: 1}, {unique: true});`) which allows this to work.